### DISCUSSION TITLE
How can I implement Guided Grad-CAM to generate fine-grained and high-res map?

### DISCUSSION THREAD

**Discussion by TommyZihao** [frgfm/torch-cam#189] [Q&A]
How can I implement Guided Grad-CAM to generate fine-grained and high-res map?

**Comment by frgfm** [comment]
Hello @TommyZihao :wave:

I'll try to answer you as exhaustively as possible:
- a raw CAM has the same spatial size as the output feature map of the selected layer (for 224x224 input on most classification models, the last conv layer will yield a 7x7 spatial size). You can then upsample that to get a higher resolution but it is upsampled. Check https://github.com/frgfm/torch-cam#quick-tour for an example
- now if you select an earlier layer, the CAM has a higher resolution. You can either do this or combine the CAM of several layers (layer fusion similar to LayerCAM). For an example, go to https://huggingface.co/spaces/frgfm/torch-cam, upload a picture, select "LayerCAM" and try "Analyze" with 3 different target layer values: "layer4" (last layer stage), "layer3" (penultimate layer stage) and "layer3+layer4" (layer fusion).

Let me know if things are still unclear :)
Cheers :v:
